Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/locate.h

```diff
@@ -0,0 +1,1886 @@
+barycentric_coordinates(const P& p, const P& q, const P& r, const P& query)
+{
+  typedef typename CGAL::Kernel_traits<P>::type                     Kernel;
+
+  return barycentric_coordinates<Kernel, P>(p, q, r, query, Kernel());
+}
+
+// Random locations
+
+/// \brief Returns a random point over the halfedge `hd`, as a `Face_location`.
+///
+/// \details The point is on the halfedge, meaning that all its barycentric coordinates
+///          are positive.
+///
+/// \tparam TriangleMesh A model of `FaceGraph`
+///
+/// \param hd a halfedge of `tm`
+/// \param tm a triangulated surface mesh
+/// \param rnd optional random number generator
+///
+template <typename TriangleMesh>
+typename internal::Locate_types<TriangleMesh>::Face_location
+random_location_on_halfedge(typename boost::graph_traits<TriangleMesh>::halfedge_descriptor hd,
+                            const TriangleMesh& tm,
+                            CGAL::Random& rnd = get_default_random())
+{
+  typedef typename internal::Locate_types<TriangleMesh>::FT               FT;
+
+  CGAL_precondition(CGAL::is_triangle_mesh(tm));
+
+  FT t(rnd.uniform_real(0., 1.));
+  return locate_in_face(hd, t, tm);
+}
+
+/// \brief Returns a random point over the face `fd`, as a `Face_location`.
+///
+/// \details The point is on the face, meaning that all its barycentric coordinates
+///          are positive.
+///
+/// \tparam TriangleMesh A model of `FaceGraph`
+///
+/// \param fd a face of `tm`
+/// \param tm a triangulated surface mesh
+/// \param rnd optional random number generator
+///
+template <typename TriangleMesh>
+typename internal::Locate_types<TriangleMesh>::Face_location
+random_location_on_face(typename boost::graph_traits<TriangleMesh>::face_descriptor fd,
+                        const TriangleMesh& CGAL_precondition_code(tm),
+                        CGAL::Random& rnd = get_default_random())
+{
+  typedef typename internal::Locate_types<TriangleMesh>::FT               FT;
+
+  CGAL_precondition(CGAL::is_triangle_mesh(tm));
+  CGAL_precondition(fd != boost::graph_traits<TriangleMesh>::null_face());
+
+  // calling 'rnd.uniform real' with double in case FT comes from an EPECK kernel (which doesn't seem to work too well)
+  FT u(rnd.uniform_real(0., 1.));
+  FT v(rnd.uniform_real(0., double(FT(1) - u)));
+
+  return std::make_pair(fd, CGAL::make_array(u, v, FT(FT(1) - u - v)));
+}
+
+/// \brief Returns a random point over the mesh `tm`.
+///
+/// \details The returned location corresponds to a random point on a random face
+///          of the mesh. The barycentric coordinates of the point in the face
+///          are thus all positive.
+///
+/// \tparam TriangleMesh A model of `FaceGraph`
+///
+/// \param tm a triangulated surface mesh
+/// \param rnd optional random number generator
+///
+template <typename TriangleMesh>
+typename internal::Locate_types<TriangleMesh>::Face_location
+random_location_on_mesh(const TriangleMesh& tm, CGAL::Random& rnd = get_default_random())
```

Comment:
Shall there is a note about the fact that the random nest is purely combinatoric (i.e. does not depend on the area of the faces)?